### DISCUSSION TITLE
Loader not triggered when routes changes

### DISCUSSION THREAD

**Discussion by WreewanMorhee** [remix-run/remix#9898] [Q&A]
I don't know how to describe it, whether it's good design or bad.
The situation I encounter is below, two routes: aa.bb.tsx  aa.bb.cc.tsx

aa.bb.tsx
```
import { Link, Outlet } from '@remix-run/react'

export const loader = async () => {
  console.warn(11111)
  return {}
}

export default function Index() {
  return (
    <>
      aa bb
      <Link to="/aa/bb/cc">Go to C</Link>
      <Outlet />
    </>
  )
}
```

aa.bb.cc.tsx

```
import { Link } from '@remix-run/react'

export const loader = async () => {
  console.warn(22222)
  return {}
}

export default function Index() {
  return (
    <>
      <Link to={'/aa/bb'}>Go Back</Link>
    </>
  )
}

```

from /aa/bb click Link to /aa/bb/cc   console 22222  --> expected 
But
from /aa/bb/cc click Link to /aa/bb  no any console --> why ? I expected console 11111

First Question:
1. If I want to print 11111, how should I achieve this ?   ( cc need to inherit the layout of /aa/bb )
2. why design such thing ? And how could this apply in what situation ? Because in my opnion or imagine: from A routes to B routes, loader of B should be called

**Comment by RafalFilipek** [comment]
I think this is because Remix is trying to optimize data loading on navigation. In your case, you have a parent and child route. Moving from `/aa/bb/cc` to `/aa/bb` in general should not affect `/aa/bb` so there is no reason to reload data.

You can use [shouldRevalidate](https://remix.run/docs/en/main/route/should-revalidate) function to force data reload.

Example: https://stackblitz.com/edit/remix-run-remix-o9qk89
